internal/commands/build/compose.go

```diff
@@ -0,0 +1,65 @@
+package build
+
+import (
+	"fmt"
+	"path"
+	"reflect"
+
+	"github.com/docker/app/types"
+	"github.com/docker/buildx/build"
+	"github.com/docker/cli/cli/compose/loader"
+	compose "github.com/docker/cli/cli/compose/types"
+)
+
+// parseCompose do parse app compose file and extract buildx Options
+// We don't rely on bake's ReadTargets + TargetsToBuildOpt here as we have to skip environment variable interpolation
+func parseCompose(app *types.App, options buildOptions) (map[string]build.Options, error) {
+	// Fixme can have > 1 composes ?
+	parsed, err := loader.ParseYAML(app.Composes()[0])
+	if err != nil {
+		return nil, err
+	}
+
+	services, err := Load(parsed)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to parse compose file: %s", err)
+	}
+
+	var zeroBuildConfig ImageBuildConfig
+	opts := map[string]build.Options{}
+	for _, service := range services {
+		if reflect.DeepEqual(service.Build, zeroBuildConfig) {
```

Comment:
Reflect package should be avoided, could we check in a function an empty build? `ImageBuildConfig` doesn't look like it super huge. Maybe something like this is enough?
```go
func checkEmptyBuild(config ImageBuildConfig) bool{
    return config.Context == "" && config.Dockerfile == "" && len(config.Args) == 0
}
```

Reply:
This is a copy/paste from bake (https://github.com/docker/buildx/blob/master/bake/compose.go#L47)
as the compose go model do use directly embedded BuildConfig references, not pointers. Just can use a `*ImageBuildConfig` so we have a nil pointer when no build configuration was parsed from yaml.